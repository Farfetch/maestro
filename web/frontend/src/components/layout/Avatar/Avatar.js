import { Avatar as AntAvatar } from "antd";
import PropTypes from "prop-types";

import { colors } from "../../../lib/colors";

const Avatar = ({ user }) => {
  let avatarText = user.email[0];
  const [firstName, lastName] = user.name.split(" ");
  if (firstName && lastName) {
    avatarText = firstName[0] + lastName[0];
  }

  return (
    <AntAvatar
      style={{
        backgroundColor: colors.orange[5]
      }}
    >
      {avatarText.toUpperCase()}
    </AntAvatar>
  );
};
Avatar.propTypes = {
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  })
};

export default Avatar;
