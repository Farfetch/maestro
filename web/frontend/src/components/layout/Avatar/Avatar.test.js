import { render } from "@testing-library/react";
import moment from "moment";
import React from "react";

import Avatar from "./Avatar";

describe("components/layout/Avatar", () => {
  test(`should render Avatar with first letter from email`, async () => {
    const user = {
      id: "user-id",
      name: "",
      email: "test@maestro.net",
      createdAt: moment("2018-05-24T13:48:04.313000"),
      updatedAt: moment("2018-05-24T13:48:04.313000")
    };

    const rendered = render(<Avatar user={user} />);

    const avatarComponent = rendered.container;
    expect(avatarComponent.outerHTML).toMatchSnapshot();
  });

  test(`should render Avatar with initials from name`, async () => {
    const user = {
      id: "user-id",
      name: "Test User",
      email: "email@maestro.net",
      createdAt: moment("2018-05-24T13:48:04.313000"),
      updatedAt: moment("2018-05-24T13:48:04.313000")
    };

    const rendered = render(<Avatar user={user} />);

    const avatarComponent = rendered.container;
    expect(avatarComponent.outerHTML).toMatchSnapshot();
  });
});
