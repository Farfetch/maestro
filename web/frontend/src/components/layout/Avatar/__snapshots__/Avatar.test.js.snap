// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Avatar should render Avatar with first letter from email 1`] = `"<div><span style=\\"background-color: rgb(250, 140, 22);\\" class=\\"ant-avatar ant-avatar-circle\\"><span class=\\"ant-avatar-string\\" style=\\"transform: scale(1) translateX(-50%);\\">T</span></span></div>"`;

exports[`components/layout/Avatar should render Avatar with initials from name 1`] = `"<div><span style=\\"background-color: rgb(250, 140, 22);\\" class=\\"ant-avatar ant-avatar-circle\\"><span class=\\"ant-avatar-string\\" style=\\"transform: scale(1) translateX(-50%);\\">TU</span></span></div>"`;
